Index: sys/arch/evbsh3/evbsh3/machdep.c
===================================================================
RCS file: /cvsroot/src/sys/arch/evbsh3/evbsh3/machdep.c,v
retrieving revision 1.75
diff -u -r1.75 machdep.c
--- sys/arch/evbsh3/evbsh3/machdep.c	19 Jun 2014 13:20:13 -0000	1.75
+++ sys/arch/evbsh3/evbsh3/machdep.c	16 Aug 2020 05:14:24 -0000
@@ -308,6 +308,7 @@
 		kernend, atop(IOM_RAM_BEGIN + IOM_RAM_SIZE),
 		VM_FREELIST_DEFAULT);
 
+	printf("uvm_page_physload succeeded!\n");
 	/* Initialize proc0 u-area */
 	sh_proc0_init();
 
Index: sys/arch/sh3/include/scifreg.h
===================================================================
RCS file: /cvsroot/src/sys/arch/sh3/include/scifreg.h,v
retrieving revision 1.10
diff -u -r1.10 scifreg.h
--- sys/arch/sh3/include/scifreg.h	18 Feb 2006 00:41:32 -0000	1.10
+++ sys/arch/sh3/include/scifreg.h	16 Aug 2020 05:14:24 -0000
@@ -36,7 +36,11 @@
 #define SH3_SCIF0_BASE	0xa4000150
 #define SH3_SCIF1_BASE	0xa4000140
 
+#ifdef SH4A
+#define SH4_SCIF_BASE	0xffe00000
+#else /* !SH4A */
 #define SH4_SCIF_BASE	0xffe80000
+#endif
 
 #ifdef SH3
 
Index: sys/uvm/uvm_physseg.c
===================================================================
RCS file: /cvsroot/src/sys/uvm/uvm_physseg.c,v
retrieving revision 1.8
diff -u -r1.8 uvm_physseg.c
--- sys/uvm/uvm_physseg.c	19 Mar 2017 23:30:39 -0000	1.8
+++ sys/uvm/uvm_physseg.c	16 Aug 2020 05:14:26 -0000
@@ -250,7 +250,7 @@
 	ps = uvm_physseg_alloc(sizeof (struct uvm_physseg));
 	if (ps == NULL) {
 		printf("uvm_page_physload: unable to load physical memory "
-		    "segment\n");
+		    "segmentX\n");
 		printf("\t%d segments allocated, ignoring 0x%"PRIxPADDR" -> 0x%"PRIxPADDR"\n",
 		    VM_PHYSSEG_MAX, pfn, pfn + pages + 1);
 		printf("\tincrease VM_PHYSSEG_MAX\n");
@@ -768,7 +768,7 @@
 
 	if (vm_nphysmem == VM_PHYSSEG_MAX) {
 		printf("uvm_page_physload: unable to load physical memory "
-		    "segment\n");
+		    "segmentY\n");
 		printf("\t%d segments allocated, ignoring 0x%llx -> 0x%llx\n",
 		    VM_PHYSSEG_MAX, (long long)start, (long long)end);
 		printf("\tincrease VM_PHYSSEG_MAX\n");
@@ -1195,6 +1195,8 @@
 	struct uvm_physseg *ps;
 	uvm_physseg_t upm;
 
+	printf("uvm_page_physload called with start:0x%"PRIxPADDR" end:0x%"PRIxPADDR" avail_start:0x%"PRIxPADDR" avail_end:0x%"PRIxPADDR" free_list:%d\n", start, end, avail_start, avail_end, free_list);
+
 	if (__predict_true(uvm.page_init_done == true))
 		panic("%s: unload attempted after uvm_page_init()\n", __func__);
 	if (uvmexp.pagesize == 0)
